protobuf/line_protocol.proto

```diff
@@ -0,0 +1,120 @@
+syntax = "proto3";
+
+/*========================================================================*/
+/*                          ENUM DEFINTIONS                               */
+/*========================================================================*/
+// Definition of triggering mode: 
+enum Activation {
+  POLLING = 0;  //=> activation only on request (default)
+  EVENT = 1;    //=> activation only on events 
+  COMPLEX = 2;  //=> combination of Polling and Event-driven mode
+}
+// Definition of digital pin modes
+enum Digital_Mode{
+  INPUT = 0; // (default)
+  OUTPUT = 1;
+  INPUT_PULLUP = 2;
+}
+// Definition of digital pin output
+enum Digital_Output{
+  LOW = 0; // (default)
+  HIGH = 1;
+}
+
+// Definition of possible UART-TTL ports
+enum UART_Port{
+  UART2 = 0; // (default)
+  UART3 = 1;
+}
+
+/*========================================================================*/
+/*                  MAIN MESSAGES: REQUEST + FEEDBACK                     */
+/*========================================================================*/
+// message sent from controller to gateway
+message Feedback {
+  //int32 code = 1;
+  uint32 profile_id = 1;  //  used to identify profile
+  string message = 2;
+  bytes data = 3;
+}
+
+// message sent from gateway to controller
+message Request {
+  // action or registration
+  oneof request_type{
+    Action  action = 1;
+    Registration  registration = 2;
+  }
+}
+
+/*========================================================================*/
+/*                  SUB MESSAGES: ACTION + REGISTRATION                   */
+/*========================================================================*/
+
+// Actions: request on existing/registered field devices
+message Action{
+
+  uint32 profile_id = 1;        // unique identification of profile
+  Activation activation = 2;  // which mode is used for the profile => polling, event-driven or combination of both
```

Comment:
Isn't the activation type already registered in the MCU during the registration process; i.e., redundant?

Reply:
Yes, you're right. I'll change that.